Sources/General/Commands/Setup.swift

```diff
@@ -0,0 +1,322 @@
+            help: .init(stringLiteral:
+                "Fetch templates from specified github repo." +
+                " Format: \"<github>\\ [branch]\"."))
+    var githubPath: String = Constants.defaultTemplatesGithub
+
+    @Option(name: [.customLong("global"), .customShort("g")],
+            help: "If specified loads templates into user home directory")
+    var shouldLoadGlobally: Bool = false
+
+    private lazy var specFactory: SpecFactory = .init()
+    private lazy var fileHelper: FileHelper = .default
+
+    // MARK: - Lifecycle
+
+    func run() throws {
+        let url = try getGitRepoPath()
+        try loadSetupFilesFromPath(url)
+    }
+
+    // MARK: - Private
+
+    private func loadSetupFilesFromPath(_ path: String) throws {
+        print("Loading setup files from \(githubPath)...")
+        let archiveURL = try downloadArchive(at: path)
+        let folderURL = try unzipArchive(at: archiveURL)
+        let setupFiles = loadSetupFiles(in: folderURL)
+        let destination = getTemplatesDestination()
+
+        var moved = [FileInfo]()
+        var isSpecModified = false
+        do {
+            moved = try move(setupFiles.templates, to: destination)
+            isSpecModified = updateSpecIfNeeded(templateURL: setupFiles.spec)
+        }
+        catch {
+            try remove(folderURL)
+            throw error
+        }
+        try remove(folderURL)
+        print()
+        displayResult(moved, isSpecModified: isSpecModified)
+    }
+
+    private func getGitRepoPath() throws -> String {
+        let components = githubPath.split(separator: " ")
+        guard let name = components.first else {
+            throw Error.githubName(githubPath)
+        }
+        var branch = "master"
+        if components.count > 1 {
+            branch = String(components[1])
+        }
+
+        return "https://github.com/\(name)/archive/\(branch).zip"
+    }
+
+    private func downloadArchive(at path: String) throws -> URL {
+        guard let url = URL(string: path) else {
+            throw Error.url(path)
+        }
+        let data = try self.data(at: url)
+        let tmpFolderURL = try tempFolderWithRandomName()
+        let fileURL = tmpFolderURL + url.lastPathComponent
+        do {
+            try data.write(to: fileURL)
+        }
+        catch {
+            throw Error.write(tmpFolderURL)
+        }
+        return fileURL
+    }
+
+    private func data(at url: URL) throws -> Data {
+        do {
+            return try Data(contentsOf: url)
+        }
+        catch {
+            throw Error.download(url)
+        }
+    }
+
+    private func tempFolderWithRandomName() throws -> URL {
+        let folderName = "\(Constants.generalTmpFolderPrefix).\(UUID().uuidString)"
+        let url = URL(fileURLWithPath: Constants.tmpFolderPath) + folderName
+        if fileHelper.createDirectory(at: url) == false {
+            throw Error.write(url)
+        }
+        return url
+    }
```

Comment:
It already uses system tmp folder. 
And storing downloaded archive in cache doesn't make much sense, since it cannot really be reused - every time a new one must be downloaded.

Reply:
https://github.com/rosberry/general/pull/1#issuecomment-689436814